lib/reversetunnel/api.go

```diff
@@ -82,16 +88,18 @@ func (params DialParams) String() string {
 //
 // There are two implementations of this interface: local and remote sites.
 type RemoteSite interface {
-	// DialAuthServer returns a net.Conn to the Auth Server of a site.
-	DialAuthServer() (net.Conn, error)
+	// DialAuthServer returns a net.Conn to the Auth Server of a site,
+	// useTunnel indicates whether connection went directly or through tunnel.
+	DialAuthServer(DialParams) (conn net.Conn, useTunnel bool, err error)
 	// Dial dials any address within the site network, in terminating
 	// mode it uses local instance of forwarding server to terminate
-	// and record the connection
-	Dial(DialParams) (net.Conn, error)
-	// DialTCP dials any address within the site network,
-	// ignores recording mode and always uses TCP dial, used
-	// in components that need direct dialer.
-	DialTCP(DialParams) (net.Conn, error)
+	// and record the connection.
+	// useTunnel indicates whether connection went directly or through tunnel.
+	Dial(DialParams) (conn net.Conn, useTunnel bool, err error)
+	// DialTCP dials any address within the site network and
+	// ignores recording mode, used in components that need direct dialer.
+	// useTunnel indicates whether connection went directly or through tunnel.
+	DialTCP(DialParams) (conn net.Conn, useTunnel bool, err error)
```

Comment:
Yeah, you're right, we don't need it. In the beginning it was needed because sending was done outside of local site, but now need for it collapses and the chain is not required anymore 👍  I've removed it from all unneeded places.